tests/validator/validator.cpp

```diff
@@ -1878,6 +1882,7 @@ TEST(Validator, unitIllDefinedEquivalentUnits)
         "CellML identifiers must contain one or more basic Latin alphabetic characters.",
         "Variable 'v2' does not have a valid units attribute.",
         "Variable 'v1' has units of '' and an equivalent variable 'v2' with non-matching units of ''. The mismatch is: ",
+        "Variables 'v1' and 'v2' do not have the same unit reduction.",
```

Comment:
I think that if we're going to add a statement like this to the error message, it should ideally come before the detailed statement telling the same thing (which is above it).  It's probably also part of a wider improvement of our error messages, but returning a check for matching units when no units are specified doesn't really make sense either?  I'd rather have the validator stop checking at that point so that the user isn't given loads of messages when the only real problem is that they haven't specified any units?